discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py

```diff
@@ -0,0 +1,220 @@
+import ipaddress
+import os
+import shutil
+import json
+from munch import Munch
+from distutils.dir_util import copy_tree
+import logging
+from test_infra import utils
+from test_infra import virsh_cleanup
+from test_infra import consts
+from test_infra.controllers.node_controllers.libvirt_controller import LibvirtController
+
+
+class TerraformController(LibvirtController):
+
+    def __init__(self, **kwargs):
+        super(TerraformController, self).__init__(**kwargs)
+        self.cluster_name = kwargs.get('CLUSTER_NAME', f'{consts.CLUSTER_PREFIX}')
+        self.params = self.terraform_params(**kwargs)
+        self.tf_folder = self.create_tf_folder()
+        self.image_path = kwargs["ISO_DOWNLOAD_PATH"]
+
+    def __del__(self):
+        self.destroy_all_nodes()
+
+    def create_tf_folder(self):
+        tf_folder = utils.get_tf_folder(self.cluster_name)
+        logging.info("Creating %s as terraform folder", tf_folder)
+        utils.recreate_folder(tf_folder)
+        copy_tree(consts.TF_TEMPLATE, tf_folder)
+        return tf_folder
+
+    # TODO move all those to conftest and pass it as kwargs
+    def terraform_params(self, **kwargs):
```

Comment:
No need to define defaults both here and in `conftest.py`

Reply:
not sure, thought about it but still did it.